forgerock-authenticator/src/main/java/org/forgerock/android/authenticator/Account.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.authenticator;
+
+import org.forgerock.android.authenticator.util.SortedList;
+
+import java.util.List;
+
+/**
+ * Account model represents an identity for the user with an issuer. It is possible for a user to
+ * have multiple accounts provided by a single issuer, however it is not possible to have multiple
+ * accounts from with same issuer for the same account name.
+ */
+public class Account extends ModelObject<Account> {
+
+    /** Unique identifier of Account */
+    private final String id;
+    /** Issuer of the account */
+    private final String issuer;
+    /** AccountName, or Username of the account for the issuer */
+    private final String accountName;
+    /** URL of Account's logo image */
+    private final String imageURL;
+    /** HEX Color code in String for Account */
+    private final String backgroundColor;
+    /** An array of Mechanism associated with current Account */
+    private final List<Mechanism> mechanisms;
+
+    private static final String TAG = Account.class.getSimpleName();
+
+    /**
+     * Creates Account object with given information
+     * @param issuer String value of issuer
+     * @param accountName String value of accountName or username
+     * @param imageURL URL of account's logo image (optional)
+     * @param backgroundColor String HEX code of account's background color (optional)
+     */
+    public Account(String issuer, String accountName, String imageURL, String backgroundColor) {
```

Comment:
Highly recommend using builder pattern with Lombok plugin, it can remove a lot of boilerplate code, like getter, setter, equal, hashcode, and maintenance is easier, when u add an attribute, no implementation is required for getter, equals.....

Reply:
Yes, I am thinking on adopt Lombok, as it is used in fr-auth. I may use Builder to construct those objects as well.